tests/unit/test__http.py

```diff
@@ -213,3 +215,16 @@ def test_api_request_conditional_retry_failed(self):
             retry=conditional_retry_mock,
         )
         http.request.assert_called_once()
+
+    def test_mtls(self):
+        client = object()
+
+        conn = self._make_one(client, api_endpoint=None)
+        self.assertEqual(conn.ALLOW_AUTO_SWITCH_TO_MTLS_URL, True)
+        self.assertEqual(conn.API_BASE_URL, "https://storage.googleapis.com")
+        self.assertEqual(conn.API_BASE_MTLS_URL, "https://storage.mtls.googleapis.com")
+
+        conn = self._make_one(client, api_endpoint="http://foo")
+        self.assertEqual(conn.ALLOW_AUTO_SWITCH_TO_MTLS_URL, False)
+        self.assertEqual(conn.API_BASE_URL, "http://foo")
+        self.assertEqual(conn.API_BASE_MTLS_URL, "https://storage.mtls.googleapis.com")
```

Comment:
Can you test the call to `client._connection.get_api_base_url_for_mtls` without using a mock?

Reply:
The reason is that `get_api_base_url_for_mtls` is only available for python-cloud-core >= 1.6.0 which we cannot upgrade it for now. We use mock here so the unit test won't break when we upgrade it in the future. Moreover, it is already well tested in python-cloud-core.